nx/lib/nx/backend.ex

```diff
@@ -218,8 +219,7 @@ defmodule Nx.Backend do
     case data do
       <<0xFFF0000000000000::64-native>> -> "-Inf"
       <<0x7FF0000000000000::64-native>> -> "Inf"
-      <<0x7FF0000000000001::64-native>> -> "NaN"
-      <<0x7FF8000000000001::64-native>> -> "NaN"
+      <<nan::64-native>> when is_nan64(nan) -> "NaN"
```

Comment:
I think a more efficient version would be something like:

    <<0x7FF::12-native, _>> -> "NaN"
    <<0xFFF::12-native, _>> -> "NaN"

No guards. :)

Reply:
Wait, wait. That won't work. We will need to check if it is little or big endian.

Reply:
You will need something like this:

```elixir
if System.endianness() == :little do
  defp inspect_float(data, 64) do
    case data do
      <<0xFFF0000000000000::64-native>> -> "-Inf"
      <<0x7FF0000000000000::64-native>> -> "Inf"
      <<_::52, 0x7FF::12-native>> -> "NaN"
      <<_::52, 0xFFF::12-native>> -> "NaN"
      <<x::float-64-native>> -> Float.to_string(x)
    end
  end
else
  defp inspect_float(data, 64) do
    case data do
      <<0xFFF0000000000000::64-native>> -> "-Inf"
      <<0x7FF0000000000000::64-native>> -> "Inf"
      <<0x7FF::12-native, _::52>> -> "NaN"
      <<0xFFF::12-native, _::52>> -> "NaN"
      <<x::float-64-native>> -> Float.to_string(x)
    end
  end
end
```